### PR TITLE
Remove all use of GNOME-specific libraries from GnomePlatform

### PR DESCRIPTION
**Pull request by directhex**
GnomePlatform relies on some *super* deprecated libraries, from old versions of GNOME 2. All of the things we use GNOME# for can be done better using more modern replacements, e.g. GIO (part of modern GLib). It'd be nice to have some Gtk+ 2.14+ methods, but that's another story for another day.

For now, this greatly simplifies the dependency chain on Linux, and makes it easier to handle some FlatPak cases in future work.

### CODE REVIEW

File: main/src/addins/GnomePlatform/Gio.cs [hunk 1]
```diff
@@ -142,6 +148,29 @@ public static string GetGSettingsString(string schema, string key)
...
+			IntPtr native = GLib.Marshaller.StringToPtrGStrdup (filename);
+			IntPtr gfile = g_file_new_for_path (native);
+			GLib.Marshaller.Free (native);
+			IntPtr native_attrs = GLib.Marshaller.StringToPtrGStrdup ("standard::icon");
+			IntPtr error;
+			IntPtr info = g_file_query_info (gfile, native_attrs, 0, IntPtr.Zero, out error);
+			if (error != IntPtr.Zero)
+				return null;
+			IntPtr icon = g_file_info_get_icon (info);
+			string[] iconid = GLib.Marshaller.Utf8PtrToString (g_icon_to_string (icon)).Split (' ');
```
Comment (iainx) [inline main/src/addins/GnomePlatform/Gio.cs]: Does the return of g_icon_to_string need to be freed here?

Comment (directhex) [inline main/src/addins/GnomePlatform/Gio.cs]: .................................................................yes.

Comment (Therzok) [inline main/src/addins/GnomePlatform/Gio.cs]: Use PtrToStringGFree.

File: main/src/addins/GnomePlatform/Gio.cs [hunk 2]
```diff
@@ -124,6 +134,43 @@ public static System.Collections.Generic.IList<DesktopApplication> GetAllForType
...
+		{
+			if (String.IsNullOrEmpty (filename))
+				return null;
+			IntPtr native = GLib.Marshaller.StringToPtrGStrdup (filename);
+			IntPtr gfile = g_file_new_for_path (native);
+			GLib.Marshaller.Free (native);
+			IntPtr native_attrs = GLib.Marshaller.StringToPtrGStrdup ("standard::icon");
+			IntPtr error;
+			IntPtr info = g_file_query_info (gfile, native_attrs, 0, IntPtr.Zero, out error);
+			if (error != IntPtr.Zero)
```
Comment (Therzok) [inline main/src/addins/GnomePlatform/Gio.cs]: Free the error if it's not IntPtr.Zero, via `g_error_free`

File: main/src/addins/GnomePlatform/Gio.cs [hunk 3]
```diff
@@ -124,6 +134,43 @@ public static System.Collections.Generic.IList<DesktopApplication> GetAllForType
...
+				return null;
+			IntPtr icon = g_file_info_get_icon (info);
```
Comment (Therzok) [inline main/src/addins/GnomePlatform/Gio.cs]: https://developer.gnome.org/gio/stable/GFileInfo.html#g-file-info-get-icon

We don't own the icon.

File: main/src/addins/GnomePlatform/Gio.cs [hunk 4]
```diff
@@ -124,6 +134,43 @@ public static System.Collections.Generic.IList<DesktopApplication> GetAllForType
...
+			string[] iconid = GLib.Marshaller.Utf8PtrToString (g_icon_to_string (icon)).Split (' ');
+			g_object_unref (icon);
```
Comment (Therzok) [inline main/src/addins/GnomePlatform/Gio.cs]: https://developer.gnome.org/gio/stable/GFileInfo.html#g-file-info-get-icon

We don't own the icon.

File: main/src/addins/GnomePlatform/Gio.cs [hunk 5]
```diff
@@ -142,6 +150,32 @@ public static string GetGSettingsString(string schema, string key)
...
 		}
 
+		public static string GetIconIdForFile (string filename)
+		{
+			if (String.IsNullOrEmpty (filename))
+				return null;
+			IntPtr native = GLib.Marshaller.StringToPtrGStrdup (filename);
+			IntPtr gfile = g_file_new_for_path (native);
+			GLib.Marshaller.Free (native);
+			IntPtr native_attrs = GLib.Marshaller.StringToPtrGStrdup ("standard::icon");
```
Comment (iainx) [inline main/src/addins/GnomePlatform/Gio.cs]: native_attrs needs freed

**Pull request merged**